### PR TITLE
Marks 2.0

### PR DESCRIPTION
**Pull request by matklad**
The main benefit here is that we no longer need to declare marks. The main drawback is that this is ~~glorious~~ horrible. WDYT?

### CODE REVIEW

File: crates/test_utils/src/mark.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+#[macro_export]
+macro_rules! _hit {
+    ($ident:ident) => {{
+        #[cfg(test)]
+        {
+            extern "C" {
+                #[no_mangle]
+                static $ident: std::sync::atomic::AtomicUsize;
```
Comment (flodiebold) [inline crates/test_utils/src/mark.rs:38]: can we add some prefix to the name, in case someone does `mark::hit!(main)` or something like that?

**Comment by flodiebold** [comment]
I like it! :smile:

File: crates/test_utils/src/mark.rs (hunk 1 above)
Comment (matklad) [reply crates/test_utils/src/mark.rs:38]: I think not without proc_macros, as we'd either need `concat_idents`, which doesn't work, or `#[link_name = stringify!(prefix $ident)]` which doesn't work either.

Comment (matklad) [reply crates/test_utils/src/mark.rs:38]: And yeah, `hit!(main)` segfaults :-( 

I guess that's fine for `#[cfg(test)]` code.

Comment (edwin0cheng) [reply crates/test_utils/src/mark.rs:38]: How about using `line! + file! + col!` macro with `stringify` similar to common C macro tricks ?

**Pull request merged**